src/commands/dataconnect-sql-setup.ts

```diff
@@ -0,0 +1,132 @@
+import { Command } from "../command";
+import { Options } from "../options";
+import { needProjectId } from "../projectUtils";
+import { pickService } from "../dataconnect/fileUtils";
+import { FirebaseError } from "../error";
+import { requireAuth } from "../requireAuth";
+import { requirePermissions } from "../requirePermissions";
+import { ensureApis } from "../dataconnect/ensureApis";
+import {
+  type SchemaMetaData,
+  SchemaSetupStatus,
+  brownfieldSqlSetup,
+  greenFieldSchemaSetup,
+  setupBrownfieldAsGreenfield,
+  iamUserIsCSQLAdmin,
+  firebaseowner,
+  DEFAULT_SCHEMA,
+  getSchemaMetaData,
+} from "../gcp/cloudsql/permissions";
+import { setupIAMUsers } from "../gcp/cloudsql/connect";
+import { logger } from "../logger";
+import { getIdentifiers } from "../dataconnect/schemaMigration";
+import { confirm } from "../prompt";
+import * as clc from "colorette";
+
+// Sets up all FDC roles (owner, writer, and reader).
+// Granting roles to users is done by the caller.
+// Returns true if schema is setup as greenfield
+export async function setupSQLPermissions(
+  instanceId: string,
+  databaseId: string,
+  schemaInfo: SchemaMetaData,
+  options: Options,
+  silent: boolean = false,
+): Promise<boolean> {
+  const schema = schemaInfo.name;
+  // Step 0: Check current user can run setup and upsert IAM / P4SA users
+  logger.info(`Attempting to Setup SQL schema "${schema}".`);
+  const userIsCSQLAdmin = await iamUserIsCSQLAdmin(options);
+  if (!userIsCSQLAdmin) {
+    throw new FirebaseError(
+      `Missing required IAM permission to setup SQL schemas. SQL schema setup requires 'roles/cloudsql.admin' or an equivalent role.`,
+    );
+  }
+  await setupIAMUsers(instanceId, databaseId, options);
+
+  if (schemaInfo.setupStatus === SchemaSetupStatus.GreenField) {
+    const rerunSetup = await confirm({
+      message: clc.yellow(
+        "Seems like the database is already setup. Would you like to rerun the setup process?",
+      ),
+      default: false,
+    });
+    if (rerunSetup) {
+      await greenFieldSchemaSetup(instanceId, databaseId, schema, options, silent);
+      logger.info(clc.green("Database setup complete."));
+      return true;
+    }
```

Comment:
If `!rerunSetup`, should we abort?

Honestly, don't find this prompt that valuable.

If they run `sql:setup`, the intent to setup SQL is already clear. It feels redundant to ask them again.

Logging is enough here.